integration-tests/jms/common/src/test/java/io/quarkus/it/artemis/jms/common/ArtemisJmsHelper.java

```diff
@@ -69,4 +69,40 @@ public void testRollback(String endpoint, JMSContext context, String queueName)
             Assertions.assertNull(message);
         }
     }
+
+    public void sendAndVerifyXACommit(JMSContext context, String queueName, String xaEndpoint, String endpoint) {
+        String body = createBody();
+        try (JMSContext autoClosedContext = context) {
+            context.createProducer().send(context.createQueue(queueName), body);
+        }
+
+        // Consume the message in xa transaction
+        Response response = RestAssured.with().get(xaEndpoint);
+        Assertions.assertEquals(jakarta.ws.rs.core.Response.Status.OK.getStatusCode(), response.statusCode());
+        Assertions.assertEquals(body, response.getBody().asString());
+
+        // Receive from queue again to confirm nothing is received i.e. message was consumed
+        // and now there is no message in the queue
+        response = RestAssured.with().get(endpoint);
+        Assertions.assertEquals(jakarta.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
```

Comment:
Is this common practice in our code? I would not recommend checking against a 500, it could be anything.

Reply:
@middagj see https://github.com/quarkiverse/quarkus-artemis/pull/320#discussion_r1380056746. With the current setup, there is a "dirty" way to work around this (detect the `null`, return an empty `String`). The clean way, however, would be through a dedicated `Exception`, and add a `NO_CONTENT` mapper for this `Exception`.

Reply:
GMTA

Reply:
Playing around a bit with retunring `null`. Since our endpoints return `String`s and not `Response`s, chances are good that we already get a `NO_CONTENT` when a `null` is returned. So this change should be easy. But I think I want a larger refactoring on the tests as well, changing to `RestAssured`s `when()...then()...` feature.